test/SlowTests/Issues/RavenDB-22036.cs

```diff
@@ -0,0 +1,142 @@
+using System.Linq;
+using FastTests;
+using Raven.Client;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations.Indexes;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_22036 : RavenTestBase
+{
+    public RavenDB_22036(ITestOutputHelper output) : base(output)
+    {
+    }
+    
+    [RavenFact(RavenTestCategory.Indexes)]
+    public void TestIfSideBySideIndexIsCreatedOnResetWithQueryParam()
```

Comment:
Can we add sharded db as well? To make sure it works correctly

Reply:
Added tests for sharded databases